espree.js

```diff
@@ -2634,6 +2847,8 @@ function parsePrimaryExpression() {
             expr = astNodeFactory.createLiteralFromSource(scanRegExp(), source);
         }
         peek();
+    } else if (allowTemplateStrings && type === Token.Template) {
```

Comment:
Ditto. Since you can only get a `Token.Template` token when template strings are enabled, you don't need to check both conditions. If the token is `Token.Template`, then you know template strings are enabled.